app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -19,28 +19,35 @@ def new
     end
 
     def show
-      if use_totp?
-        show_totp_prompt
+      if current_otp_method == :totp
+        render :confirm_totp
       else
         @phone_number = user_decorator.masked_two_factor_phone_number
       end
     end
 
     def confirm
-      @phone_number = user_decorator.masked_two_factor_phone_number
-      @code_value = current_user.direct_otp if FeatureManagement.prefill_otp_codes?
+      if current_otp_method == :totp
+        render :confirm_totp
```

Comment:
I don't think this is needed. A user cannot select a TOTP option unless they are TOTP enabled, in which case it is automatically chosen for them. In other words, if you are TOTP enabled, you automatically see the `confirm_otp` screen when you sign in. You can then select to receive an SMS or phone call instead. If you click on one of those phone options, there is nothing on the screen that lets you choose the TOTP option.